aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -109,44 +126,98 @@ module aptos_framework::transaction_validation {
     fun prologue_common(
         sender: &signer,
         gas_payer: &signer,
-        txn_sequence_number: u64,
+        replay_protector: ReplayProtector,
         txn_authentication_key: Option<vector<u8>>,
         txn_gas_price: u64,
         txn_max_gas_units: u64,
         txn_expiration_time: u64,
         chain_id: u8,
         is_simulation: bool,
     ) {
+        let sender_address = signer::address_of(sender);
+        let gas_payer_address = signer::address_of(gas_payer);
         assert!(
             timestamp::now_seconds() < txn_expiration_time,
             error::invalid_argument(PROLOGUE_ETRANSACTION_EXPIRED),
         );
         assert!(chain_id::get() == chain_id, error::invalid_argument(PROLOGUE_EBAD_CHAIN_ID));
 
+        // TODO[Orderless]: Here, we are mantaining the same order of validation steps as before orderless txns were introduced.
+        // Ideally, do the replay protection check in the end after the authentication key check and gas payment checks.
+
+        // Check for replay protection
+        match (replay_protector) {
+            SequenceNumber(txn_sequence_number) => {
+                check_for_replay_protection_regular_txn(
+                    sender,
+                    gas_payer,
+                    txn_sequence_number,
+                );
+            },
+            Nonce(nonce) => {
+                check_for_replay_protection_orderless_txn(
+                    sender_address,
+                    nonce,
+                    txn_expiration_time,
+                );
+            }
+        };
+
+        // Check if the authentication key is valid
+        if (option::is_some(&txn_authentication_key)) {
+            assert!(
+                txn_authentication_key == option::some(account::get_authentication_key(sender_address)),
+                error::invalid_argument(PROLOGUE_EINVALID_ACCOUNT_AUTH_KEY),
+            );
+        } else {
+            assert!(
+                allow_missing_txn_authentication_key(sender_address),
+                error::invalid_argument(PROLOGUE_EINVALID_ACCOUNT_AUTH_KEY)
+            );
+        };
+
+        // Check if the gas payer has enough balance to pay for the transaction
+        let max_transaction_fee = txn_gas_price * txn_max_gas_units;
+        if (!skip_gas_payment(
```

Comment:
it seems missing `!features::transaction_simulation_enhancement_enabled()` piece, is it intentional?

Reply:
`transaction_simulation_enhancement_enabled` flag was enabled 6 months ago. It's default to true now. So, just simplified the statement. 
The prologue code is part of the state. So, this is backward compatible.

Reply:
can we then remove features::transaction_simulation_enhancement_enabled() from transaction_validation.move completely (i.e. assume it is true throughout?)

if you want that in separate PR, sounds good too.

Reply:
Done.

Reply:
The previous run passed: https://github.com/aptos-labs/aptos-core/actions/runs/14673892408
I'm running again now: https://github.com/aptos-labs/aptos-core/actions/runs/14762270024